xtuner/v1/train/trainer.py

```diff
@@ -892,11 +914,7 @@ def _maybe_save(self, is_snapshot: bool = False) -> bool:
 
         meta_path = self.work_dir / self._META_PATH
 
-        optimizer_path = (
-            checkpoint_path / self._SAVE_OPTIMIZER_DIR
-            if self._resume_cfg.load_optimizer_states or self._resume_cfg.load_optimizer_args
-            else None
-        )
+        optimizer_path = checkpoint_path / self._SAVE_OPTIMIZER_DIR
```

Comment:
Do we need to check the `load_checkpoint_cfg` here? Otherwise optimizer will always be saved